nipype/pipeline/engine/nodes.py

```diff
@@ -367,7 +368,22 @@ def run(self, updatehash=False):
             except:
                 os.remove(hashfile_unfinished)
                 raise
-            shutil.move(hashfile_unfinished, hashfile)
+
+            t = time()
+            timeout = float(self._config['execution']['job_finished_timeout'])
+            timed_out = True
+            while True:
+                try:
+                    shutil.move(hashfile_unfinished, hashfile)
```

Comment:
maybe we can apply the EAFP principle here? Even though theoretically this shouldn't require a timeout?